wemake_python_styleguide/violations/consistency.py

```diff
@@ -314,3 +316,47 @@ class MultipleIfsInComprehensionViolation(ASTViolation):
     #: Error message shown to the user.
     error_template = 'Found list comprehension with multiple `if`s'
     code = 307
+
+
+class BadNumberSuffixViolation(TokenizeViolation):
+    """
+    Forbids to use capital (``XOBE``) in numbers.
+
+    Reasoning:
+        It is possible to write ``0xFF`` in two different ways:
```

Comment:
Reasoning right now do not look like real reasoning. 
I guess it should be something like: 
```
 Octal, hex, binary and scientific notation suffixes could be written in two possible notations: lowercase and uppercase. Which brings confusion and decreases code consistency and readability. 
```

That is better in my opinion.